### ISSUE TITLE
Unexpected output occurs when using set_ with a path that is an integer between version 7.06 and later.

### ISSUE THREAD

**Issue by huypxgear** [dgilland/pydash#226]
The output differs when using set_ between version 7.0.6 and later.
Example:
```
result = {}
set_(result, ['a', 1], {})
print(result)
```
Version 7.0.6:
Output: 
`{'a': {'1': {}}}`
Version > 7.0.6
Output: 
`{'a': [None, {}]}`

Does it intend to change the nested value from a dictionary to a list? I don't see any changelogs that refer to the output behavior for `set_` in the recent update.

**Comment by dgilland** [comment]
The change was intentional in order to fix #213.

The choice for whether to treat an integer key value as a list index or a dictionary key isn't always clear cut and, unfortunately in this case it could go either way (i.e. one person may want it to be a dictionary key whereas another person may want it to be a list index).

For cases where the default behavior of `set_` isn't desirable, I would suggest using [pydash.set_with](https://pydash.readthedocs.io/en/latest/api.html#pydash.objects.set_with):

```python
>>> from pydash import set_with
>>> set_with({}, ['a', 1], {}, lambda: {})
{'a': {1: {}}}
```

The customizer callback to `set_with` can take 0-3 arguments (the number of arguments passed to the callback is based on the callback's function signature). The callback arguments are `(nested_value, key, nested_object)` and is called for each key in the object path being set. This allows for a specific object type to be created when it the target object's key is unset.

**Comment by huypxgear** [comment]
Thank for your information
